### DISCUSSION TITLE
How to provide request / response operation examples having common parameters

### DISCUSSION THREAD

**Discussion by forik** [microsoft/typespec#4695] [OpenAPI3] [answered]
I have the following typespec code:
```
import "@typespec/http";

using TypeSpec.Http;
@service({
  title: "Widget Service",
})
namespace DemoService;

model Widget {
  @visibility("read", "update")
  @path
  id: string;

  weight: int32;
  color: "red" | "blue";
}

model CommonParameters {
  @header("x-request-id")
  requestId: string;
  @header("x-api-version")
  apiVersion?: string;
}

@error
model Error {
  code: int32;
  message: string;
}

@route("/widgets")
@tag("Widgets")
interface Widgets {
  @opExample(
    #{
      parameters: #{w: #{id: "", weight: 1, color: "red"}}, 
      returnType: #{id: "", weight: 1, color: "red"}
    }, 
    #{ title: "widget example", description: "Widget example desc" }
  )
  @post create(@body w: Widget, ...CommonParameters): Widget | Error;
}
```

This is how error looks like:
<img width="836" alt="image" src="https://github.com/user-attachments/assets/b92fd249-1f5d-4185-80f6-39e27cc7d36b">

Having CommonParameters.requestId as a required header I'm not able to provide an example which describes only a body, but compiler requires me to provide the fields from CommonParameters model. Is the a way to work this out? Thanks

**Comment by timotheeguerin** [comment, marked as answer]
opExample represent the cannonical example of your operation. Your operation takes requestId as a required parameter so it should be provided in the example as well.
